File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        meaning = {
            'I':1,
            'V':5,
            'X':10,
            'L':50,
            'C':100,
            'D':500,
            'M':1000
        }
        sum=0
        previous=0
        for i in range(len(s)-1,-1,-1):
            current = meaning[s[i]]
            if current < previous:
                sum-=meaning[s[i]]
            else:
                sum+=meaning[s[i]]
            previous=current
        return sum         

File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<Character,Integer> meaning = new HashMap<>();
        meaning.put('I',1);
        meaning.put('V',5);
        meaning.put('X',10);
        meaning.put('L',50);
        meaning.put('C',100);
        meaning.put('D',500);
        meaning.put('M',1000);
        int sum=0;
        int previous=0;
        int current=0;
        for (int i=s.length()-1;i>=0;i--){
            current = meaning.get(s.charAt(i));
            if (current < previous){
                sum-=current;
            }else{
                sum+=current;
            }
            previous=current;
        }
        return sum;
    }
}